beacon-chain/core/peerdas/helpers.go

```diff
@@ -0,0 +1,248 @@
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/interfaces"
+	"github.com/prysmaticlabs/prysm/v5/crypto/hash"
+	"github.com/prysmaticlabs/prysm/v5/encoding/bytesutil"
+	ethpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+)
+
+const (
+	// Number of field elements per extended blob
+	fieldElementsPerExtBlob = 2 * cKzg4844.FieldElementsPerBlob
+
+	// Bytes per cell
+	bytesPerCell = cKzg4844.FieldElementsPerCell * cKzg4844.BytesPerFieldElement
+
+	// Number of cells in the extended matrix
+	extendedMatrixSize = fieldparams.MaxBlobsPerBlock * cKzg4844.CellsPerExtBlob
+)
+
+type (
+	extendedMatrix []cKzg4844.Cell
+
+	cellCoordinate struct {
+		blobIndex uint64
+		cellID    uint64
+	}
+)
+
+var (
+	errCustodySubnetCountTooLarge          = errors.New("custody subnet count larger than data column sidecar subnet count")
+	errCellNotFound                        = errors.New("cell not found (should never happen)")
+	errCurveOrder                          = errors.New("could not set bls curve order as big int")
+	errBlsFieldElementNil                  = errors.New("bls field element is nil")
+	errBlsFieldElementBiggerThanCurveOrder = errors.New("bls field element higher than curve order")
+	errBlsFieldElementDoesNotFit           = errors.New("bls field element does not fit in BytesPerFieldElement")
+)
+
+// https://github.com/ethereum/consensus-specs/blob/dev/specs/_features/eip7594/das-core.md#helper-functions
+func custodyColumns(nodeId enode.ID, custodySubnetCount uint64) (map[uint64]bool, error) {
+	dataColumnSidecarSubnetCount := params.BeaconConfig().DataColumnSidecarSubnetCount
+
+	// Compute the custodied subnets.
+	subnetIds, err := CustodyColumnSubnets(nodeId, custodySubnetCount)
+	if err != nil {
+		return nil, errors.Wrap(err, "custody subnets")
+	}
+
+	columnsPerSubnet := cKzg4844.CellsPerExtBlob / dataColumnSidecarSubnetCount
+
+	// Knowing the subnet ID and the number of columns per subnet, select all the columns the node should custody.
+	// Columns belonging to the same subnet are contiguous.
+	columnIndices := make(map[uint64]bool, custodySubnetCount*columnsPerSubnet)
+	for i := uint64(0); i < columnsPerSubnet; i++ {
+		for subnetId := range subnetIds {
+			columnIndex := dataColumnSidecarSubnetCount*i + subnetId
+			columnIndices[columnIndex] = true
+		}
+	}
+
+	return columnIndices, nil
+}
+
+func CustodyColumnSubnets(nodeId enode.ID, custodySubnetCount uint64) (map[uint64]bool, error) {
+	dataColumnSidecarSubnetCount := params.BeaconConfig().DataColumnSidecarSubnetCount
+
+	// Check if the custody subnet count is larger than the data column sidecar subnet count.
+	if custodySubnetCount > dataColumnSidecarSubnetCount {
+		return nil, errCustodySubnetCountTooLarge
+	}
+
+	// First, compute the subnet IDs that the node should participate in.
+	subnetIds := make(map[uint64]bool, custodySubnetCount)
+	i := uint64(0)
+
+	for uint64(len(subnetIds)) < custodySubnetCount {
```

Comment:
Can you make `i` part of the for loop, separating out the way it is done is odd

Reply:
Fixed in https://github.com/prysmaticlabs/prysm/pull/13877/commits/039f9eb7e950e98b12172664843dda22b0270175.